### PR TITLE
Re-construct: Fixed issue #604

### LINKED ISSUE #604: Can't remove a member's nickname and roles with guild_edit_member
**Git commit reference**
`d5f49c3fe22d1d14cadc1500f67fa9bdcffebca0`

**Describe the bug**
When using `cluster::guild_edit_member` to remove the nickname or roles of a member, it doesn't remove them. No error thrown. 

**To Reproduce**
e.g. this code:
```c++
auto m = dpp::find_guild_member(825407338755653642, 189759562910400512);
m.nickname = "";
m.roles = {};
bot.guild_edit_member(m);
```

**Expected behavior**
To remove the roles and nickname

**System Details:**
 - OS: Ubuntu
 - Discord Client used for testing [desktop]

**the fix**
The way the build_json method is made, the nickname and roles fields won't be send to discord when they're empty.

### PR DESCRIPTION
**Pull request by harshfeudal**
Fix the bug (#604) of not removing roles and reset the nickname. Reset the old PR because affiliated with the old pull req.

### CODE REVIEW

File: src/dpp/guild.cpp [hunk 1]
```diff
@@ -229,13 +229,20 @@ std::string guild_member::build_json(bool with_id) const {
 			j["communication_disabled_until"] = json::value_t::null;
 		}
 	}
-	if (!this->nickname.empty())
-		j["nick"] = this->nickname;
+	
+	if (!this->nickname.empty()) {
+		j["nick"] = json::value_t::null;
+	} else {
+		j["nick"] = find_user(user_id)->username;
```
Comment (Commandserver) [inline src/dpp/guild.cpp:236]: This should set the nickname from the passed member object, not from cache

File: src/dpp/guild.cpp [hunk 2]
```diff
@@ -229,13 +229,20 @@ std::string guild_member::build_json(bool with_id) const {
 			j["communication_disabled_until"] = json::value_t::null;
 		}
 	}
-	if (!this->nickname.empty())
-		j["nick"] = this->nickname;
+	
+	if (!this->nickname.empty()) {
```
Comment (Commandserver) [inline src/dpp/guild.cpp:233]: You sure this works? You clear the nickname if it's not empty, and set it, if it's empty? accidentally negated i guess

**Review by Commandserver — CHANGES_REQUESTED** [review]

File: src/dpp/guild.cpp (hunk 1 above)
Comment (harshfeudal) [reply src/dpp/guild.cpp]: i have updated the new one

File: src/dpp/guild.cpp (hunk 2 above)
Comment (harshfeudal) [reply src/dpp/guild.cpp:217]: Oopps

**Review by Commandserver — APPROVED** [review]

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**